Userland/Libraries/LibJS/SourceTextModule.cpp

```diff
@@ -19,18 +81,571 @@ Result<NonnullRefPtr<SourceTextModule>, Vector<Parser::Error>> SourceTextModule:
     if (parser.has_errors())
         return parser.errors();
 
-    // FIXME: Implement the rest of ParseModule.
-    return adopt_ref(*new SourceTextModule(realm, filename, move(body)));
+    // 3. Let requestedModules be the ModuleRequests of body.
+    auto requested_modules = module_requests(*body);
+
+    // 4. Let importEntries be ImportEntries of body.
+    Vector<ImportEntry> import_entries;
+    for (auto const& import_statement : body->imports())
+        import_entries.extend(import_statement.entries());
+
+    // 5. Let importedBoundNames be ImportedLocalNames(importEntries).
+    // Note: Since we have to potentially extract the import entry we just use importEntries
+    //       In the future it might be an optimization to have a set/map of string to speed up the search.
+
+    // 6. Let indirectExportEntries be a new empty List.
+    Vector<ExportEntry> indirect_export_entries;
+
+    // 7. Let localExportEntries be a new empty List.
+    Vector<ExportEntry> local_export_entries;
+
+    // 8. Let starExportEntries be a new empty List.
+    Vector<ExportEntry> star_export_entries;
+
+    // Note: Not in the spec but makes it easier to find the default.
+    RefPtr<ExportStatement> default_export;
+
+    // 9. Let exportEntries be ExportEntries of body.
+    // 10. For each ExportEntry Record ee of exportEntries, do
+    for (auto const& export_statement : body->exports()) {
+
+        if (export_statement.is_default_export()) {
+            VERIFY(!default_export);
+            VERIFY(export_statement.entries().size() == 1);
+            VERIFY(export_statement.has_statement());
+
+            auto const& entry = export_statement.entries()[0];
+            VERIFY(entry.kind == ExportStatement::ExportEntry::Kind::LocalExport);
+            VERIFY(import_entries.find_if(
+                                     [&](ImportEntry const& import_entry) {
+                                         return import_entry.local_name == entry.local_or_import_name;
+                                     })
+                       .is_end());
+            default_export = export_statement;
+        }
+
+        for (auto const& export_entry : export_statement.entries()) {
+            // FIXME: Store default entry
```

Comment:
Could also be a bit clearer IMO.

Reply:
This FIXME was already resolved just didn't remove the comment